src/redux/services/projectConfigMonitor.ts

```diff
@@ -0,0 +1,79 @@
+import {FSWatcher, watch} from 'chokidar';
+import {readFileSync} from 'fs';
+import {sep} from 'path';
+
+import {ProjectConfig} from '@models/appconfig';
+
+import {updateProjectConfig} from '@redux/reducers/appConfig';
+import {AppDispatch} from '@redux/store';
+
+let watcher: FSWatcher;
+
+/**
+ * Creates a monitor for the specified folder and dispatches folder events using the specified dispatch
+ */
+
+export function monitorProjectConfigFile(dispatch: AppDispatch, filePath?: string | null) {
+  if (!filePath && watcher) {
+    watcher.close();
+    return;
+  }
+  if (watcher) {
+    watcher.close();
+  }
+
+  const absolutePath = `${filePath}${sep}.monokle`;
+
+  watcher = watch(absolutePath, {
+    persistent: true,
+    usePolling: true,
+    interval: 1000,
+  });
+
+  watcher
+    .on('add', () => {
+      readApplicationConfigFileAndUpdateProjectSettings(absolutePath, dispatch);
+    })
+    .on('change', () => {
+      readApplicationConfigFileAndUpdateProjectSettings(absolutePath, dispatch);
+    })
+    .on('unlink', () => {
+      readApplicationConfigFileAndUpdateProjectSettings(absolutePath, dispatch);
+    });
+
+  watcher
```

Comment:
Is there any reason why this cannot be moved to the object on line 33?

Reply:
Nope, there is no specific reason - will merge them